be/src/vec/common/allocator.cpp

```diff
@@ -177,7 +177,9 @@ void Allocator<clear_memory_, mmap_populate, use_mmap>::throw_bad_alloc(
 
 template <bool clear_memory_, bool mmap_populate, bool use_mmap>
 void* Allocator<clear_memory_, mmap_populate, use_mmap>::alloc(size_t size, size_t alignment) {
-    doris::thread_context()->large_memory_check = false;
+    if (doris::enable_thread_catch_bad_alloc) {
+        doris::thread_context()->large_memory_check = false;
```

Comment:
not modify this line. maybe we could delete large_memory_check, it should always be true.

Reply:
done